inst/@sym/private/mat_replace.m

```diff
@@ -66,12 +79,7 @@
     end
   end
 
-  if (length(subs) == 1 && islogical(subs{1}))
-    %% A(logical) = B
-    z = mat_mask_asgn(A, subs{1}, b);
-    return
-
-  elseif (length(subs) == 1 && strcmp(subs{1}, ':') && length(b) == 1)
```

Comment:
I have remove the first if as that condition is now taken care at the start of this method.